shared/node.go

```diff
@@ -159,6 +159,18 @@ func (node *Node) handleEvent(message *messaging.PocketEnvelope) error {
 	switch contentType {
 	case messaging.NodeStartedEventType:
 		log.Println("[NOOP] Received NodeStartedEvent")
+	case messaging.BeforeHeightChangedEventType:
+		log.Println("Received BeforeNewHeightEvent")
+		node.GetBus().GetP2PModule().HandleEvent(message.Content)
+	case messaging.AddressBookAtHeightEventType:
+		log.Printf("Received AddressBookAtHeightEvent: %v\n", message.Content)
+		// DISCUSS (deblasis): here we'd delegate a module to handle the event.
+		// We need an open PersistenceRWContext where we can update the validator list (sounds like churn management).
```

Comment:
There is some confusion here.

The validator list is part of the world state (i.e. it is part of the persistence layer). When you apply a block, you should be able to say "Give me validators at height N from the SQL engine".

The address book is a list of all actors (staked actors, full nodes, light clients, etc...) that the current node can connect to. For example, the `serviceURL` of a validator can be matched with the address book so we know their active IP.

Reply:
You basically answered my questions/clarified my doubts with this.

Since the P2P addressBook is just going to live in memory, the `peersManager` should suffice and therefore I believe that some of the events I introduced can be revisited/removed. I'll update and re-request review accordingly.